Code/Common/include/sitkCompositeTransform.h

```diff
@@ -0,0 +1,204 @@
+ * included in the transformation during the optimization.
+ *
+ * Optimization Flags:
+ * The m_TransformsToOptimize flags hold one flag for each transform in the
+ * queue, designating if each transform is to be used for optimization. Note
+ * that all transforms in the queue are applied in TransformPoint, regardless
+ * of these flags states'. The methods GetParameters, SetParameters,
+ * ComputeJacobianWithRespectToParameters, GetTransformCategory,
+ * GetFixedParameters, and SetFixedParameters all query these
+ * flags and include only those transforms whose corresponding flag is set.
+ * Their input or output is a concatenated array of all transforms set for use
+ * in optimization. The goal is to be able to optimize multiple transforms at
+ * once, while leaving other transforms fixed. See the above example.
+ *
+ * Setting Optimization Flags:
+ * A transform's optimization flag is set when it is added to the queue, and
+ * remains set as other transforms are added. The methods
+ * SetNthTransformToOptimize* and SetAllTransformToOptimize* are used to
+ * set and clear flags arbitrarily. SetOnlyMostRecentTransformToOptimizeOn is
+ * a convenience method for setting only the most recently added transform
+ * for optimization, with the idea that this will be a common practice.
+ *
+ * Indexing:
+ * The index values used in GetNthTransform and
+ * SetNthTransformToOptimize* and SetAllTransformToOptimize* follow the
+ * order in which transforms were added. Thus, the first transform added is at
+ * index 0, the next at index 1, etc.
+ *
+ * Inverse:
+ * The inverse transform is created by retrieving the inverse from each
+ * sub transform and adding them to a composite transform in reverse order.
+ * The m_TransformsToOptimizeFlags is copied in reverse for the inverse.
+ *
+ * \ingroup ITKTransform
+ */
+class SITKCommon_EXPORT CompositeTransform
+    : public Transform {
+public:
+  using Self = CompositeTransform;
+  using Superclass = Transform;
+
+  /** \brief Construct an empty CompositeTransform
+   *
+   *  The created CompositeTransform is initialized with zero transforms.
+   *  Additional transforms of dimensions can be added.
+   */
+  explicit CompositeTransform(unsigned int dimensions);
+
+  /** \brief Create CompositeTransform converted or holding the transform argument.
+   *
+   * If the Transform is internally a CompositeTransform, a shallow copy to
+   * the internal transform will be made. Otherwise a new CompositeTransform is
+   * constructed which holds the transform argument.
+   */
+  CompositeTransform( const Transform &);
+
+  /** \brief A shallow copy constructor */
+  CompositeTransform( const CompositeTransform & );
+
+  ~CompositeTransform() override;
+
+  CompositeTransform &operator=( const CompositeTransform &);
+
+    /** Name of this class */
+  std::string GetName( ) const override  { return std::string("CompositeTransform");}
+
+
+  /** \brief Remove nested composite transforms
```

Comment:
I don't think it should say "Remove nested...", as I believe the operation, flattens internal composite transforms, they aren't removed.